src/Lynx.purs

```diff
@@ -0,0 +1,276 @@
+module Lynx where
+
+import Prelude
+
+import Data.Const (Const)
+import Data.Either.Nested (type (\/))
+import Data.Foldable (fold, foldMap, for_)
+import Data.Functor.Coproduct.Nested (type (<\/>))
+import Data.Map (Map)
+import Data.Map as Data.Map
+import Data.Maybe (Maybe(..))
+import Data.Maybe as Data.Maybe
+import Effect.Aff.Class (class MonadAff)
+import Halogen as H
+import Halogen.Component.ChildPath (cp1, cp2, cp3)
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Lynx.Data.Expr (EvalError(..), Expr, ExprType, Key, boolean_, cents_, datetime_, print, reflectType, string_, toArray, toBoolean, toCents, toDateTime, toObject, toPair, toString)
+import Lynx.Data.Form (Field, Input(..), InputSource(..), Page, Section, ValidationError(..), asyncFromTypeahead, errorsToArray, getValue)
+import Lynx.Data.Form as Lynx.Data.Form
+import Network.RemoteData (RemoteData(..), fromEither)
+import Ocelot.Block.Button as Button
+import Ocelot.Block.Card as Card
+import Ocelot.Block.FormField as FormField
+import Ocelot.Block.Format as Format
+import Ocelot.Block.Input as Input
+import Ocelot.Block.ItemContainer (boldMatches)
+import Ocelot.Block.Layout as Layout
+import Ocelot.Block.Toggle (toggle) as Toggle
+import Ocelot.Component.DateTimePicker as DateTimePicker
+import Ocelot.Component.Dropdown as Dropdown
+import Ocelot.Component.Dropdown.Render as Dropdown.Render
+import Ocelot.Component.Typeahead as Typeahead
+import Ocelot.Data.Currency (parseCentsFromDollarStr)
+import Ocelot.HTML.Properties (css)
+
+type State =
+  { form :: RemoteData EvalError { expr :: Page Expr, evaled :: Page ExprType }
+  , values :: Map Key ExprType
+  }
+
+data Query a
+  = EvalForm (Page Expr) a
+  | UpdateValue Key (InputSource ExprType) a
+  | DropdownQuery Key (Dropdown.Message Query ExprType) a
+  | DateTimePickerQuery Key DateTimePicker.Message a
+  | TypeaheadSingleQuery  Key (Typeahead.Message Query Maybe ExprType) a
+
+type ParentInput = RemoteData EvalError (Page Expr)
```

Comment:
I'm not sure about this one. Makes sense for `RemoteData` to come from the parent, but `EvalError` comes from `Lynx`. Although I don't think I'd want to nest our data structure to something like `RemoteData e (Either EvalError (Page Expr))` 😅

What do you think about something like `EvalError e` being an open variant, so we could do something like:
```purescript
type ParentInput e = RemoteData (EvalError e) (Page Expr)
```

Reply:
Another option is to separate concerns and handle failure of `RomeData` in the parent, and only load Lynx on success, passing only `Page Expr` down as input. This might make more sense, as different parents might want to handle failure differently.

Reply:
Good point. `EvalError` is completely an implementation detail of lynx (that will hopefully go away sooner rather than later).

I think making input just `Page Expr` seems the most reasonable.